assets/src/stories-editor/blocks/amp-story-page-attachment/edit.js

```diff
@@ -0,0 +1,301 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, RawHTML } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { InspectorControls, RichText } from '@wordpress/block-editor';
+import { select } from '@wordpress/data';
+import {
+	SelectControl,
+	PanelBody,
+	Button,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import './edit.css';
+import { getBackgroundColorWithOpacity } from '../../../common/helpers';
+import { PostSelector } from '../../components';
+
+class PageAttachmentEdit extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			selectedPost: null,
+			isOpen: false,
+			searchValue: '',
+			failedToFetch: false,
+		};
+
+		this.toggleAttachment = this.toggleAttachment.bind( this );
+		this.removePost = this.removePost.bind( this );
+	}
+
+	componentDidMount() {
+		this.fetchSelectedPost();
+	}
+
+	componentWillUnmount() {
+		this.isStillMounted = false;
+	}
+
+	componentDidUpdate( prevProps ) {
+		const {
+			attributes,
+			backgroundColor,
+			customBackgroundColor,
+			textColor,
+			setAttributes,
+		} = this.props;
+
+		if ( attributes.postId !== prevProps.attributes.postId ) {
+			this.fetchSelectedPost();
+		}
+
+		if (
+			backgroundColor !== prevProps.backgroundColor ||
+			customBackgroundColor !== prevProps.customBackgroundColor ||
+			textColor !== prevProps.textColor
+		) {
+			const { style, attachmentClass } = this.getWrapperAttributes();
+			const newAttributes = { wrapperStyle: style };
+			if ( textColor !== prevProps.textColor ) {
+				newAttributes.attachmentClass = attachmentClass;
+			}
+			setAttributes( newAttributes );
+		}
+	}
+
+	fetchSelectedPost() {
+		const { postId } = this.props.attributes;
+		this.isStillMounted = true;
+		if ( postId ) {
+			const fetchRequest = this.fetchRequest = apiFetch( {
+				path: `/wp/v2/posts/${ postId }`,
```

Comment:
Should we allow other post types?

Reply:
That sounds reasonable to me, yes. For example, personally I would rather want to embed a page here than a post.

Reply:
Hello @miina @swissspidy 
This is Gary from Taiwan, I am a marketer. 
If you please consider adding embed function, that way for some of the people who have product page can insert a link directly without extra works.

Reply:
> If you please consider adding embed function, that way for some of the people who have product page can insert a link directly without extra works.

Note that embedding will be possible from the same site only for now. The post can be easily searched by title or content and then selected with just one click, the content will be pulled in without extra work.